instaloader/__main__.py

```diff
@@ -307,6 +320,8 @@ def _main(instaloader: Instaloader, targetlist: List[str],
         else:
             # Instaloader did not do anything
             instaloader.context.log("usage:" + usage_string())
+        exit_code = ExitCodes.NON_FATAL_ERROR
```

Comment:
This leads to a NON_FATAL_ERROR exit code when using Instaloader just for logging in or importing session cookies, which I think is a valid and non-error use case. I think the exit_code should be only set in the `else`, if Instaloader was not used for logging in and the usage was printed.

Reply:
This is one case in which I wasn't sure what was the best way to handle (see the last note in the main text of the PR).

So I guess in the else branch (wasn't logged in) an error should be returned (in this case I think initialization failure is more appropriate), but if it was logged in then success is returned?

Reply:
I think calling Instaloader only for creating a session file is a legit non-error use case if successful. When calling Instaloader without a target, only with a —login or —load-cookies option the exit code should indicate success if succesful, and LOGIN_FAILURE otherwise.

In the other case, INIT_FAILURE is probably the best exit code.